website/docs/cloud-docs/workspaces/health.mdx

```diff
@@ -157,34 +176,30 @@ resource "aws_instance" "hashiapp" {
   subnet_id                   = aws_subnet.hashiapp.id
   vpc_security_group_ids      = [aws_security_group.hashiapp.id]
   key_name                    = aws_key_pair.generated_key.key_name
+
+  tags = {
+    Name = "hashiapp"
+  }
 }
 
 check "ami_version_check" {
+  data "aws_instance" "hashiapp_current" {
+    instance_tags = {
+      Name = "hashiapp"
+    }
+  }
+
   assert {
     condition = aws_instance.hashiapp.ami == data.hcp_packer_image.hashiapp_image.cloud_image_id
     error_message = "Must use the latest available AMI, ${data.hcp_packer_image.hashiapp_image.cloud_image_id}."
   }
 }
 ```
 
-#### Monitoring certificate expiration
+Check blocks execute as the last step of a plan or apply after Terraform has planned or provisioned your infrastructure. Each health assessment run performs a speculative plan in its normal course of operation. In practice, this means that check assertions are evaluated against a hypothetical future state where any planned changes have already been applied to the resources under management.
```

Comment:
what was the main intention of this paragraph? do we need to discuss the check blocks as part of a plan apply if we're only talking about assessment runs and not regular operations? i think the second part could use a bit of clarification. my understanding is that continuous validation isnt aware of planned changes to config, so it's only checking against existing resources.

Reply:
Since a few weeks ago, assessments now use full plan runs. So now CV is actually being done in the context of a plan/apply, and that's the reason for the confusion.